product-service/utils/db.js

```diff
@@ -0,0 +1,72 @@
+// FIXME require
+const { Client } = require('pg');
+
+const {
+  PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD,
+} = process.env;
+
+export const createClient = () => new Client({
+  host: PG_HOST,
+  port: PG_PORT,
+  database: PG_DATABASE,
+  user: PG_USERNAME,
+  password: PG_PASSWORD,
+  ssl: {
+    rejectUnauthorized: false, // to avoid warring in this example
+  },
+  connectionTimeoutMillis: 5000, // time in millisecond for termination of the database query
+});
+
+export const queryProduct = async (client, productId) => {
+  const { rows } = await client.query({
+    text: `
+    SELECT products.*, stocks.count 
+    FROM products
+    LEFT JOIN stocks ON products.id = stocks.product_id
+    WHERE products.id = $1
+  `,
+    values: [productId],
+  });
+  return rows.length ? rows[0] : null;
+};
+
+export const queryAllProducts = async (client) => {
+  const { rows } = await client.query({
+    text: `
+      SELECT products.*, stocks.count 
+      FROM products
+      LEFT JOIN stocks ON products.id = stocks.product_id
+    `,
+  });
+  return rows;
+};
+
+export const queryCreateProduct = async (client, data) => {
+  const {
+    title, description, price, count,
+  } = data;
+  try {
+    await client.query('BEGIN');
```

Comment:
I would not recommend using a shared `client` instance with transactions. Await statement passes a control to another function. Thus, a request from another function can be executed after the BEGIN statement in the same client.

Reply:
Thank you very much for the comment! I really appreciate your advice!